dask/dataframe/io/hdf.py

```diff
@@ -297,6 +297,13 @@ def get_keys_stops_divisions(path, key, stop, sorted_index, chunksize):
         """
         with pd.HDFStore(path, mode=mode) as hdf:
             keys = [k for k in hdf.keys() if fnmatch(k, key)]
+            if not keys:
+                # https://github.com/dask/dask/issues/5934
+                # TODO: remove this if/when pandas copes with all keys
+                keys = [
+                    n._v_pathname
+                    for n in hdf._handle.walk_nodes("/", classname="Table")
```

Comment:
You probably want to add
```python
if fnmatch(n._v_pathname, key)
```
here, otherwise `keys` will contain all nodes in the file which is probably too much.